src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { request } from "@octokit/request";
+
+import { getCache } from "./get-cache";
+import { getInstallationAccessToken } from "./get-installation-access-token";
+import { getSignedJsonWebToken } from "./get-signed-json-web-token";
+import LRUCache from "lru-cache";
+import { State } from "./types";
+
+interface AppOptions {
+  id: number;
+  privateKey: string;
+  baseUrl?: string;
+  cache?: LRUCache<number, string>;
+}
+
+export class App {
```

Comment:
why did you turn it into a class?

The methods are now creating a new pre-bound function each time they are called instead of once when the api is created. And the code is less readable now I think.

Reply:
Because of the way TypeScript works. It’s used like a class, and is defined as one in the definition file.

I don’t know if there is another way though. I did it this way so it would match the definition file

Reply:
I've changed up my implementation a little, it only creates the bound functions once.

Reply:
@gr2m Is this better?

Reply:
Yeah, good compromise :)